utility for generating parameter table for wiki.

yamadayy opened the pull request.

YoshiyukiYamada committed:
utility for generating parameter table for wiki.

YoshiyukiYamada committed:
Fix code climate warnings.

YoshiyukiYamada committed:
Fix code climate warnings.

YoshiyukiYamada committed:
pix is not unit. So, the parameters xxx which has unit of /pix is renamed xxx_per_pix.

YoshiyukiYamada committed:
pix is not unit. So, the parameters xxx which has unit of /pix is renamed xxx_per_pix.

YoshiyukiYamada committed:
efficiency_grid is simple number. Remove unit. (efficiency.py)
Remove comment of "why it has unit" from parameter_table.py.

yamadayy commented:
This is the code for generating parameter table. Also parameter table in E2E name space of wiki is updated.

astronasutarou commented on jasmine_toolkit/utils/parameter_table.py:
```diff
@@ -0,0 +1,111 @@
+    ["earth mu", p.earth_mu, 1],
+    ["earth C1", p.earth_c1, 1],
+    ["earth C2", p.earth_c2, 1],
+    p.EARTH_MASS,
+    p.CONST_OF_GRAVITATION,
+    p.EQUATORIAL_EARTH_RADIUS,
+    p.POLAR_EARTH_RADIUS,
+    p.ONE_YEAR,
```
これらはテーブルに出力しなくてもいいように思います．

yamadayy replied:
削除しました

astronasutarou commented on jasmine_toolkit/utils/parameter_table.py:
```diff
@@ -0,0 +1,111 @@
+def write_to_file(filename: str):
+    fp = open(filename, "w", encoding="utf-8")
+    fp.write("Parameter List\n^Name ^Value ^Units ^Comments^\n")
+    for i in range(len(v_list)):
+        if isinstance(v_list[i], JasmineConstant):
+            fp.write("|" + str(v_list[i].name) + "|" + str(v_list[i].value)
+                     + "|" + str(v_list[i].unit) + "|"
+                     + str(v_list[i].reference) + "|\n")
+        elif isinstance(v_list[i], list):
+            write_without_name_item(fp, i)
```
`v_list` と `i` は常にセットで使われているようなので `v_list` の中身で for を回したほうがシンプルにならないでしょうか？以下のようなイメージです．

``` python
for v in v_list:
  if isinstance(v, JasmineConstant):
    write_jasmine_constant(v)
  elif isinstance(v, list):
    write_without_name_item(v)
  else:
    raise ValueError(f'wrong item type {type(v)}')
```

yamadayy replied:
修正しました

astronasutarou commented on jasmine_toolkit/utils/parameter_table.py:
```diff
@@ -0,0 +1,111 @@
+
+
+if __name__ == '__main__':
+    write_to_file("tmp.txt")
```
1. 標準出力からのコピペで十分かと思いましたがファイルに書いたほうがいいでしょうか．
2. ローカルに `tmp.txt` があると上書きしてしまうのでファイル名のハードコードは避けたほうがいいと思います．ハードコードする場合には安易に上書きされないような安全装置が欲しいですね．

yamadayy replied:
出力が長いのでファイルにしました。また、これは臼井さん(かサーバーの管理権限がある人)の仕事ですが、将来は手動コピーではなくwikiの中の決まった名前のファイルに定期的に上書きしてゆくという運用になるはずなので、既存ファイルに上書きが良いと思いました。従って直していません。

astronasutarou reviewed (changes requested):
目的を考えると物理定数などは出力に含めないほうが良いかと思います．出力についてもコメントしましたが今回は別にいいかなという気もします．

YoshiyukiYamada committed:
Remove physical constants from the output. Simplify loop.(parameter_table.py)
Because the Efficiency class is fixed as the efficiency has non dimension, test_parameters.py needs to fix.

astronasutarou reviewed (approved):
Approved.
